File: tests/fixtures/kotlin/data_and_sealed.kt
package com.example.demo.model

/** Plain data class — primary-ctor components should become fields. */
data class Point(val x: Int, val y: Int) : Comparable<Point> {
    override fun compareTo(other: Point): Int = x.compareTo(other.x)
}

/** Sealed class hierarchy — permits is implicit (subclasses in same file/module). */
sealed class Shape {
    abstract fun area(): Double
}

data class Circle(val radius: Double) : Shape() {
    override fun area(): Double = Math.PI * radius * radius
}

class Square(val side: Double) : Shape() {
    override fun area(): Double = side * side
}

object UnitShape : Shape() {
    override fun area(): Double = 1.0
}

/**
 * Enum class with a primary-ctor field, value-carrying entries, and a
 * declared method that follows the entries.
 */
enum class Status(val label: String, val weight: Int) : java.io.Serializable {
    ACTIVE("Active", 3),
    INACTIVE("Inactive", 1),
    BANNED("Banned", 0),
    UNKNOWN("?", -1);

    fun display(): String = "$label[$weight]"

    companion object {
        fun parse(s: String): Status = values().firstOrNull { it.label == s } ?: UNKNOWN
    }
}

File: tests/fixtures/scala/data_and_sealed.scala
package com.example.demo.model

/** Case class — primary-ctor params surface as fields even without val/var. */
case class Point(x: Int, y: Int) extends Ordered[Point] {
  override def compare(that: Point): Int = x - that.x
}

/** Sealed trait + case class / case object branches. */
sealed trait Shape {
  def area: Double
}

case class Circle(radius: Double) extends Shape {
  override def area: Double = math.Pi * radius * radius
}

class Square(val side: Double) extends Shape {
  override def area: Double = side * side
}

case object UnitShape extends Shape {
  override def area: Double = 1.0
}

/** Scala 3 enum with primary-ctor field and a declared method. */
enum Status(val label: String, val weight: Int) extends java.io.Serializable:
  case Active   extends Status("Active",   3)
  case Inactive extends Status("Inactive", 1)
  case Banned   extends Status("Banned",   0)
  case Unknown  extends Status("?", -1)

  def display: String = s"$label[$weight]"
